packages/cli/cloud/src/create-project/utils/apply-default-name.ts

```diff
@@ -0,0 +1,22 @@
+import { DistinctQuestion } from 'inquirer';
+import { ProjectAnswers } from '../../types';
+
+export function applyDefaultName(
+  newDefaultName: string,
+  questions: ReadonlyArray<DistinctQuestion<ProjectAnswers>>,
+  defaultValues: Partial<ProjectAnswers>
+): {
+  newQuestions: ReadonlyArray<DistinctQuestion<ProjectAnswers>>;
+  newDefaultValues: Partial<ProjectAnswers>;
+} {
+  const newDefaultValues = { ...defaultValues, name: newDefaultName };
+
+  const newQuestions = questions.map((question) => {
+    if (question.name === 'name') {
+      return { ...question, default: newDefaultName };
+    }
+    return { ...question };
```

Comment:
Do we really need to create a fresh object?
```suggestion
    return question;
```

If the point was to copy the object to not keep the reference, then you might want to do a deepClone instead of destructuring. 🤔

Reply:
Went for the JSON parse/stringify to do the copy, although it'll only work with simple objects/array (The "...question" would've at least kept the functions if they were added to the question)

The only solution without any "effective" drawback would be to add a proper class or factory to instantiate the "question" and "default values" objects and use that instead. You think it's worth refactoring now?

Reply:
I think you didn't responded to my first question 😄 

But in any case, if we really want to copy, we can use https://lodash.com/docs/4.17.15#cloneDeep

Reply:
Indeed, I want to avoid having the same references. Initial code works for the current code since we only have scalar values so far.
Using deepClone now, should be better!